### PR TITLE
Create an id property, this new property will be used as unique to in…

### PR DESCRIPTION
**Pull request by dlascarez**
Create an ID property, this new property will be used as unique to identify each column instead of "data".
This change will fix the error: 

https://github.com/l-lin/angular-datatables/issues/1748

### CODE REVIEW

File: src/models/settings.ts [hunk 1]
```diff
@@ -4,6 +4,8 @@ export interface ADTSettings extends DataTables.Settings {
   columns?: ADTColumns[];
 }
 export interface ADTColumns extends DataTables.ColumnSettings {
+  /** Define a column index */
```
Comment (shanmukhateja) [inline src/models/settings.ts:7]: I think this should be "Define the column's unique identifier"

File: package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "angular-datatables",
-  "version": "16.0.0",
+  "version": "16.0.1",
```
Comment (shanmukhateja) [inline package.json:3]: Please revert this change as this PR is about bug fix.

File: src/angular-datatables.directive.ts [hunk 3]
```diff
@@ -75,6 +75,10 @@ export class DataTableDirective implements OnDestroy, OnInit {
           reject('Both the table and dtOptions cannot be empty');
           return;
         }
+        
+        // Set a column unique
+        resolvedDTOptions.columns.forEach((col: ADTColumns, i: number) => col.id = i);
```
Comment (shanmukhateja) [inline src/angular-datatables.directive.ts:80]: We need to skip all columns which have id defined by user.

File: src/models/settings.ts [hunk 4]
```diff
@@ -4,6 +4,8 @@ export interface ADTSettings extends DataTables.Settings {
...
+  id?: number;
```
Comment (shanmukhateja) [inline src/models/settings.ts:8]: Can we make this a string so user can add their own? Identifiers based on positioning can lead to weird edge cases as datatables.net allows column re-ordering and other complex functionality.

**Review by shanmukhateja — CHANGES_REQUESTED** [review]
Left feedback

File: src/models/settings.ts (hunk 1 above)
Comment (dlascarez) [reply src/models/settings.ts]: I have changed the comment

File: src/angular-datatables.directive.ts (hunk 3 above)
Comment (dlascarez) [reply src/angular-datatables.directive.ts]: To generate the random string, I have included a new method:

`private getColumnUniqueId(): string {
    let result = '';
    const characters = 'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789';

    for (let i = 0; i < 6; i++) {
      const randomIndex = Math.floor(Math.random() * characters.length);
      result += characters.charAt(randomIndex);
    }

    return result.trim();
  }`

File: src/models/settings.ts (hunk 4 above)
Comment (dlascarez) [reply src/models/settings.ts]: Done, this property has been changed from number to string.

**Review by shanmukhateja — APPROVED** [review]
Looks good to me. Need to wait for Louis's review to merge.

File: src/angular-datatables.directive.ts (hunk 3 above)
Comment (shanmukhateja) [reply src/angular-datatables.directive.ts]: We might use an external library for this in the future but it should in theory get us going for now. 

I am marking this as resolved.
